### PR TITLE
Describe Jira priorities and release blockers on Beam site

### PR DESCRIPTION
**Pull request by kennknowles**
I attempted to write up a page that corresponds to the [Jira priorities discussion](https://lists.apache.org/thread.html/7a1d91a7a0c388d28df024bd03846ca936395278d1079166bb28f170@<dev.beam.apache.org>)

I also added a page about how release blockers work. This has so far just been [implied by the release guide](https://beam.apache.org/contribute/release-guide/#4-triage-release-blocking-issues-in-jira).

http://apache-beam-website-pull-requests.storage.googleapis.com/9862/index.html

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/)
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python2/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python2_PVR_Flink_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/)
XLang | --- | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | --- | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

File: website/src/contribute/release-blocking.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+not have urgency, while Blocker/P0 is reserved for issues so urgent they do not
+wait for a release.
+
+## Release-blocking features
+
+By default, features do not block releases. Beam has a steady 6 week cadence of
+cutting release branches and releasing. Features "catch the train" or else wait
+for the next release.
+
+A feature can block a release if there is community consensus to delay a
```
Comment (Ardagan) [inline website/src/contribute/release-blocking.md]: Might be a good idea to request dev list discussion and link to it on corresponding jira ticket.

**Comment by pabloem** [comment]
I personally had not used these in that way. I used 'blocker' to block a release. I saw value in this because it's not strictly necessary to speak to the release manager to block the release. My expectation was that the release manager would know to bump the version of non-blocker issues, but wait until blockers were resolved to move forward with the release.

File: website/src/contribute/release-blocking.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+# Release blockers
+
+A release blocking Jira is any open Jira that has its `Fix Version` field set
```
Comment (aaltay) [inline website/src/contribute/release-blocking.md]: I believe fix version is used as a project planning tool and people are flexible in changing fix version dates during the first bug scrub that happens in the release cycle.

Does JIRA have a field for planning to target a version without a hard intention to block that release?

File: website/src/contribute/release-blocking.md (hunk 1 above)
Comment (aaltay) [reply website/src/contribute/release-blocking.md]: +1 to the idea here.

I recall a dev@ discussion where we agreed that it makes sense for vendors to be transparent about due dates they are targeting for features and community to take that into account for releases.

File: website/src/contribute/release-blocking.md (hunk 2 above)
Comment (kennknowles) [reply website/src/contribute/release-blocking.md]: There is no existing functionality for that. We could have a convention. For example Pablo mentioned that he makes bugs "Blocker" in order to block the release. So a bug that is not a "Blocker" but has Fix Version set could be not release blocking. But I think that is probably confusing and causes more work for the release manager.

File: website/src/contribute/release-blocking.md (hunk 1 above)
Comment (kennknowles) [reply website/src/contribute/release-blocking.md]: I like how you phrase it. We could say it here, that you should write to dev@ and be transparent about targeting features to a release, and community can listen and decide.

**Comment by kennknowles** [comment]
@pabloem that could work, too. I am happy just to have some system. I guess this should continue on the dev list since policy should not be made on a PR. But I do have a couple questions to clarify before bringing it back:

1. How do you have a P0 emergency bug if Blocker is used for release blocker? Just without Fix Version?
2. What is the use of Fix Version, just making the release manager push them to the next release? If the release manager has to check with people before moving it, then to me that is "blocking" the release anyhow, in the sense that the release does not proceed until the ticket is moved off of it.

**Comment by pabloem** [comment]
I think I withdraw my comments on this PR, so it's free to move forward as far as I'm concerned.

**Comment by kennknowles** [comment]
OK. Then I will merge. But please feel free to revise and discuss more. I just want to have something written up as a starting point and something to point to.

**Pull request merged**